packages/common-library/src/common_library/changelog.py

```diff
@@ -0,0 +1,242 @@
+    def to_string(self) -> str:
+        """Converts entry to a formatted string for documentation"""
+
+    @abstractmethod
+    def get_version(self) -> Version | None:
+        """Returns the version associated with this entry, if any"""
+
+
+class NewEndpoint(ChangelogEntryAbstract):
+    """Indicates when an endpoint was first added"""
+
+    entry_type = ChangelogType.NEW
+
+    def __init__(self, version: str):
+        self.version = version
+
+    def to_string(self) -> str:
+        return f"New in *version {self.version}*\n"
+
+    def get_version(self) -> Version:
+        return Version(self.version)
+
+
+class ChangedEndpoint(ChangelogEntryAbstract):
+    """Indicates a change to an existing endpoint"""
+
+    entry_type = ChangelogType.CHANGED
+
+    def __init__(self, version: str, message: str):
+        self.version = version
+        self.message = message
+
+    def to_string(self) -> str:
+        return f"Changed in *version {self.version}*: {self.message}\n"
+
+    def get_version(self) -> Version:
+        return Version(self.version)
+
+
+class DeprecatedEndpoint(ChangelogEntryAbstract):
+    """Indicates an endpoint is deprecated and should no longer be used"""
+
+    entry_type = ChangelogType.DEPRECATED
+
+    def __init__(self, alternative_route: str, version: str | None = None):
+        self.alternative_route = alternative_route
+        self.version = version
+
+    def to_string(self) -> str:
+        base_message = "🚨 **Deprecated**"
+        if self.version:
+            base_message += f" in *version {self.version}*"
+
+        return (
+            f"{base_message}: This endpoint is deprecated and will be removed in a future release.\n"
+            f"Please use `{self.alternative_route}` instead.\n\n"
+        )
+
+    def get_version(self) -> Version | None:
+        return Version(self.version) if self.version else None
+
+
+class RetiredEndpoint(ChangelogEntryAbstract):
+    """Indicates when an endpoint will be or was removed"""
+
+    entry_type = ChangelogType.RETIRED
+
+    def __init__(self, version: str, message: str):
+        self.version = version
+        self.message = message
+
+    def to_string(self) -> str:
+        return f"Retired in *version {self.version}*: {self.message}\n"
+
+    def get_version(self) -> Version:
+        return Version(self.version)
+
+
+def create_route_description(
+    *,
+    base: str = "",
+    changelog: Sequence[ChangelogEntryAbstract] | None = None,
+) -> str:
+    """
+    Builds a consistent route description with optional changelog information.
+
+    Args:
+        base (str): Main route description.
+        changelog (Sequence[ChangelogEntry]): List of changelog entries.
+
+    Returns:
+        str: Final description string.
+    """
+    parts = []
+
+    if base:
+        parts.append(base)
+
+    if changelog:
+        changelog_strings = [entry.to_string() for entry in changelog]
+        parts.append("\n".join(changelog_strings))
```

Comment:
are there not too many `\n`

Reply:
I would like that they look like this
![image](https://github.com/user-attachments/assets/61159099-d8c2-4e2c-9006-188bab84dd87)

Reply:
@sanderegg  I tuned it a bit. Now

![image](https://github.com/user-attachments/assets/90d4d854-3958-4388-97e8-acf074bcc78e)

Will produce

![image](https://github.com/user-attachments/assets/219e9b76-6d0d-4e4d-be50-23373b5f91cd)